### DISCUSSION TITLE
Cognitive complexity statistic

### DISCUSSION THREAD

**Discussion by ernestasjuska** [StefanMaron/BusinessCentral.LinterCop#285] [New Rule]
Hi,

While looking information on cyclomatic complexity I came across "cognitive complexity".

Info: https://www.sonarsource.com/blog/cognitive-complexity-because-testability-understandability/
Whitepaper: https://www.sonarsource.com/docs/CognitiveComplexity.pdf

It seems to describe complexity of the function better than cyclomatic complexity while remaining very simple.

Maybe this method could be implemented too?

Thanks.

**Comment by Arthurvdv** [comment]
Hi @ernestasjuska. This sounds promising and could be great rule next to the Cyclomatic Complexity (LC0010). I'll will look into this and see if I can make a new rule for this in the undefined future.

For now I've moved your issue to a discussion to have a follow-up here.

**Reply by Arthurvdv** [reply]
@ernestasjuska, The undefined future is here 🙋‍♂️

I wanted to thank you for bringing Cognitive Complexity to my attention. I read the whitepaper back in 2023 and was enthusiastic about it. I didn’t have the experience back then to create this comprehensive rule for it. Over the past years, my knowledge has grown and a [recent issue](https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/924) was a challenged and reminder for me to start building a rule for the Cognitive complexity.

It's now available in the [v0.34.1](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.34.1)  Pre-Release version of the LinterCop if you want to try it out.

**Reply by ernestasjuska** [reply]
Hi, @Arthurvdv,

That is very good to hear. I will tell others to try it too. :)

**Reply by pri-kise** [reply]
Are we able to customize the threshold in the LinterCop.json ?
https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/master/BusinessCentral.LinterCop/LinterCop.json
If this is possible it would be useful to have this documented.

**Reply by Arthurvdv** [reply]
@pri-kise, valid question! I've added this to the [documentation of the LC0090 rule](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0090#threshold) just a few minuted ago.
